### ISSUE TITLE
Database error in Contribution Pivot Chart

### ISSUE THREAD

**Issue by arthurgoldstein** [eileenmcnaughton/nz.co.fuzion.extendedreport#465]
Can you look in on this? Seems that apostrophe in a name is causing a string problem.

Get this error:
Select year(`receive_date`) as year, sum(`total_amount`) as amount from civicrm_contribution contribution LEFT JOIN civicrm_contact civicrm_contact
        ON contribution.contact_id = civicrm_contact.id
        LEFT JOIN civicrm_address address
        ON address.contact_id = civicrm_contact.id
        AND address.is_primary = 1 Where civicrm_contact.sort_name LIKE '%Aicha's Naturals%' group by year(`receive_date`) [nativecode=1064 ** You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 's Naturals%' group by year(`receive_date`)' at line 5]

When running the instructions:
Go to Reports menu » Contribution Reports.
Click the New Contribution Report buton.
Click the Extended Report - Contribution Pivot Chart link.
In the Select Row Fields drop-down, choose Contact Name.
In the Select Column Header drop-down, choose Breakdown by Year.
Press View Results.

Which are from:
https://civicrm.stackexchange.com/questions/33315/annual-donor-report/33316#33316

**Comment by arthurgoldstein** [comment]
After updating civicrm and trying to run the report get this error:
 [error] $Fatal Error Details = Array
(
    [callback] => Array
        (
            [0] => CRM_Core_Error
            [1] => handle
        )

    [code] => -19
    [message] => DB Error: no such field
    [mode] => 16
    [debug_info] => 
SELECT id, title, frontend_title, description, frontend_description, visibility, parents, saved_search_id
FROM   civicrm_group
WHERE  id IN (93,36,94,34,102,96,118,114,103,130,172,124,121,132,174,1,176,166,95,136,140,108,109,100,98,167,147,141,2,32,17,139,145,101,107,42,44,43,127,137,138,37,19,35,92,119,170,144,158,163,165,84,91,51,40,41,75,86,77,76,78,152,8,66,115,14,18,142,143,85,175,129,179,111,128,151,168,148,120,171,52,160,90,89,125,146,178,99,150,126,134,88,63,149,155,153,156,135,169,110,25,26,16,83,113,64,81,133,117,131,106,50,33,154,105,159,164,3,157,97,27,116)
 ORDER BY title [nativecode=1054 ** Unknown column 'frontend_title' in 'field list']
    [type] => DB_Error
    [user_info] => 
SELECT id, title, frontend_title, description, frontend_description, visibility, parents, saved_search_id
FROM   civicrm_group
WHERE  id IN (93,36,94,34,102,96,118,114,103,130,172,124,121,132,174,1,176,166,95,136,140,108,109,100,98,167,147,141,2,32,17,139,145,101,107,42,44,43,127,137,138,37,19,35,92,119,170,144,158,163,165,84,91,51,40,41,75,86,77,76,78,152,8,66,115,14,18,142,143,85,175,129,179,111,128,151,168,148,120,171,52,160,90,89,125,146,178,99,150,126,134,88,63,149,155,153,156,135,169,110,25,26,16,83,113,64,81,133,117,131,106,50,33,154,105,159,164,3,157,97,27,116)
 ORDER BY title [nativecode=1054 ** Unknown column 'frontend_title' in 'field list']
    [to_string] => [db_error: message="DB Error: no such field" code=-19 mode=callback callback=CRM_Core_Error::handle prefix="" info="
SELECT id, title, frontend_title, description, frontend_description, visibility, parents, saved_search_id
FROM   civicrm_group
WHERE  id IN (93,36,94,34,102,96,118,114,103,130,172,124,121,132,174,1,176,166,95,136,140,108,109,100,98,167,147,141,2,32,17,139,145,101,107,42,44,43,127,137,138,37,19,35,92,119,170,144,158,163,165,84,91,51,40,41,75,86,77,76,78,152,8,66,115,14,18,142,143,85,175,129,179,111,128,151,168,148,120,171,52,160,90,89,125,146,178,99,150,126,134,88,63,149,155,153,156,135,169,110,25,26,16,83,113,64,81,133,117,131,106,50,33,154,105,159,164,3,157,97,27,116)
 ORDER BY title [nativecode=1054 ** Unknown column 'frontend_title' in 'field list']"]
)

**Comment by eileenmcnaughton** [comment]
@arthurgoldstein have you run the civicrm upgrade script? That field should exist in your database from 5.31 on 

https://github.com/civicrm/civicrm-core/commit/35fd9d21457113beb75e2e27b82f8749717c3d64

**Comment by arthurgoldstein** [comment]
You are correct. Upgrading the database fixed that error. The original error is still there though, with the apostrophe's causing problems. Thanks, Arthur

**Comment by eileenmcnaughton** [comment]
I just dug into this report and found a whole host of issues around grouping by receive_date (which I have now fixed in master) - but none around the appostrope

![image](https://user-images.githubusercontent.com/336308/118478597-317fbe00-b764-11eb-8e47-48a006d66a59.png)
